jena-text/src/main/java/org/apache/jena/query/text/TextIndexES.java

```diff
@@ -0,0 +1,427 @@
+            Settings settings = Settings.builder()
+                    .put(CLUSTER_NAME, esSettings.getClusterName()).build();
+            List<InetSocketTransportAddress> addresses = new ArrayList<>();
+            for(String host: esSettings.getHostToPortMapping().keySet()) {
+                InetSocketTransportAddress addr = new InetSocketTransportAddress(InetAddress.getByName(host), esSettings.getHostToPortMapping().get(host));
+                addresses.add(addr);
+            }
+
+            InetSocketTransportAddress socketAddresses[] = new InetSocketTransportAddress[addresses.size()];
+            client = new PreBuiltTransportClient(settings).addTransportAddresses(addresses.toArray(socketAddresses));
+            LOGGER.debug("Successfully initialized the client");
+        }
+
+
+        IndicesExistsResponse exists = client.admin().indices().exists(new IndicesExistsRequest(INDEX_NAME)).get();
+        if(!exists.isExists()) {
+            Settings indexSettings = Settings.builder()
+                    .put(NUM_OF_SHARDS, esSettings.getShards())
+                    .put(NUM_OF_REPLICAS, esSettings.getReplicas())
+                    .build();
+            LOGGER.debug("Index with name " + INDEX_NAME + " does not exist yet. Creating one with settings: " + indexSettings.toString());
+            client.admin().indices().prepareCreate(INDEX_NAME).setSettings(indexSettings).get();
+        }
+
+
+
+    }
+
+
+    /**
+     * Constructor used mainly for performing Integration tests
+     * @param config an instance of {@link TextIndexConfig}
+     * @param client an instance of {@link TransportClient}. The client should already have been initialized with an index
+     */
+    public TextIndexES(TextIndexConfig config, Client client, String indexName) {
+        this.docDef = config.getEntDef();
+        this.isMultilingual = true;
+        this.client = client;
+        this.INDEX_NAME = indexName;
+    }
+
+    /**
+     * We do not have any specific logic to perform before committing
+     */
+    @Override
+    public void prepareCommit() {
+        //Do Nothing
+
+    }
+
+    /**
+     * Commit happens in the individual get/add/delete operations
+     */
+    @Override
+    public void commit() {
+        // Do Nothing
+    }
+
+    /**
+     * not really sure what we need to roll back.
+     */
+    @Override
+    public void rollback() {
+       //Not sure what to do here
+
+    }
+
+    /**
+     * We don't have resources that need to be closed explicitely
+     */
+    @Override
+    public void close() {
+        // Do Nothing
+
+    }
+
+    /**
+     * Update an Entity. Since we are doing Upserts in add entity anyways, we simply call {@link #addEntity(Entity)}
+     * method that takes care of updating the Entity as well.
+     * @param entity the entity to update.
+     */
+    @Override
+    public void updateEntity(Entity entity) {
```

Comment:
This is OK. However, I noticed that TextIndex.updateEntity is never actually called from within Jena - there are only addition and deletion events but no updates as such. It is possible that this method will be removed from jena-text.